src/control/server/storage/class.go

```diff
@@ -0,0 +1,262 @@
+//
+// (C) Copyright 2018-2021 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package storage
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+	"text/template"
+
+	"github.com/dustin/go-humanize"
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/common"
+	"github.com/daos-stack/daos/src/control/logging"
+)
+
+const (
+	confOut   = "daos_nvme.conf"
+	nvmeTempl = `[Nvme]
+{{ $host := .Bdev.Hostname }}{{ $tier := .Tier }}{{ range $i, $e := .Bdev.DeviceList }}    TransportID "trtype:PCIe traddr:{{$e}}" Nvme_{{$host}}_{{$i}}_{{$tier}}
+{{ end }}    RetryCount 4
+TimeoutUsec 0
+ActionOnTimeout None
+AdminPollRate 100000
+HotplugEnable No
+HotplugPollRate 0
+`
+	// device block size hardcoded to 4096
+	fileTempl = `[AIO]
+{{ $host := .Bdev.Hostname }}{{ $tier := .Tier }}{{ range $i, $e := .Bdev.DeviceList }}    AIO {{$e}} AIO_{{$host}}_{{$i}}_{{$tier}} 4096
+{{ end }}`
+	kdevTempl = `[AIO]
+{{ $host := .Bdev.Hostname }}{{ $tier := .Tier }}{{ range $i, $e := .Bdev.DeviceList }}    AIO {{$e}} AIO_{{$host}}_{{$i}}_{{$tier}}
+{{ end }}`
+	mallocTempl = `[Malloc]
+NumberOfLuns {{.Bdev.DeviceCount}}
+LunSizeInMB {{.Bdev.FileSize}}000
+`
+	gbyte   = 1000000000
+	blkSize = 4096
+
+	msgBdevNone    = "in config, no nvme.conf generated for server"
+	msgBdevEmpty   = "bdev device list entry empty"
+	msgBdevBadSize = "backfile_size should be greater than 0"
+)
+
+// bdev describes parameters and behaviors for a particular bdev class.
+type bdev struct {
+	templ   string
+	isEmpty func(*BdevConfig) string                // check no elements
+	isValid func(*BdevConfig) string                // check valid elements
+	init    func(logging.Logger, *BdevConfig) error // prerequisite actions
+}
```

Comment:
@tanabarr: Looks like we've got some conflicting approaches to solving this problem that need to be resolved. I'm still digesting this PR but wanted to make you aware of this right away.

Reply:
I did move the file from bdev to storage pkg to avoid circular dependency between packages. Bdev does include storage, so couldn't have included bdev from storage.
I don't like it but that enables me to make progress. Can refactor later.